mne_sandbox/connectivity/cfc.py

```diff
@@ -0,0 +1,564 @@
+import numpy as np
+from mne.filter import band_pass_filter
+from mne.utils import _time_mask
+from mne.time_frequency import cwt_morlet
+from mne.preprocessing import peak_finder
+from mne.utils import ProgressBar
+import mne
+import warnings
+
+
+# Supported PAC functions
+_pac_funcs = ['plv', 'glm', 'mi_tort', 'mi_canolty', 'ozkurt', 'otc']
+# Calculate the phase of the amplitude signal for these PAC funcs
+_hi_phase_funcs = ['plv']
+
+
+def phase_amplitude_coupling(inst, f_phase, f_amp, ixs, pac_func='ozkurt',
+                             ev=None, tmin=None, tmax=None, n_cycles=None,
+                             scale_amp_func=None,  return_data=False,
+                             concat_epochs=False, n_jobs=1, verbose=None):
+    """ Compute phase-amplitude coupling between pairs of signals using pacpy.
+
+    Parameters
+    ----------
+    inst : an instance of Raw or Epochs
+        The data used to calculate PAC
+    f_phase : array, dtype float, shape (2,)
+        The frequency range to use for low-frequency phase carrier.
+    f_amp : array, dtype float, shape (2,)
+        The frequency range to use for high-frequency amplitude modulation.
+    ixs : array-like, shape (n_pairs x 2)
+        The indices for low/high frequency channels. PAC will be estimated
+        between n_pairs of channels. Indices correspond to rows of `data`.
+    pac_func : string, list of strings
+        The function for estimating PAC. Corresponds to functions in pacpy.pac.
+        Must be one of ['plv', 'glm', 'mi_canolty', 'mi_tort', 'ozkurt'].
+    ev : array, shape (n_events, 3) | array, shape (n_events,) | None
+        MNE events array. To be supplied if data is 2D and output should be
+        split by events. In this case, tmin and tmax must be provided. If
+        ndim == 1, it is assumed to be event indices, and all events will be
+        grouped together.
+    tmin : float | list of floats, shape (n_pac_windows,) | None
+        If ev is not provided, it is the start time to use in inst. If ev
+        is provided, it is the time (in seconds) to include before each
+        event index. If a list of floats is given, then PAC is calculated
+        for each pair of tmin and tmax.
+    tmax : float | list of floats, shape (n_pac_windows,) | None
+        If ev is not provided, it is the stop time to use in inst. If ev
+        is provided, it is the time (in seconds) to include after each
+        event index. If a list of floats is given, then PAC is calculated
+        for each pair of tmin and tmax.
+    scale_amp_func : None | function
+        If not None, will be called on each amplitude signal in order to scale
+        the values. Function must accept an N-D input and will operate on the
+        last dimension. E.g., skl.preprocessing.scale
+    return_data : bool
+        If True, return the phase and amplitude data along with the PAC values.
```

Comment:
Can you make it more explicit, using backticks and variable order with maybe an example?